dash/include/dash/algorithm/Sort.h

```diff
@@ -0,0 +1,217 @@
+#ifndef DASH__ALGORITHM__SORT_H__
+#define DASH__ALGORITHM__SORT_H__
+
+#include <dash/iterator/GlobIter.h>
+#include <dash/algorithm/LocalRange.h>
+
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <math.h>
+#include <mpi.h>
+
+using namespace std;
+
+namespace dash {
+
+template <
+  typename ElementType,
+  class    PatternType >
+void sort(
+  /// Iterator to the initial position in the sequence
+  const GlobIter<ElementType, PatternType> & first,
+  /// Iterator to the final position in the sequence
+  const GlobIter<ElementType, PatternType> & last)
+{
+  auto index_range  = dash::local_index_range(first, last);
+  auto lbegin_index = index_range.begin;
+  auto lend_index   = index_range.end;
+  auto & pattern     = first.pattern();
+  auto lrange_begin = (first + pattern.global(lbegin_index)).local();
+  auto lrange_end   = lrange_begin + lend_index;
+
+  auto n = dash::distance(first, last);
+  auto p = dash::size();
+
+  ElementType first_possible_key = * dash::min_element (first, last);
+  ElementType last_possible_key  = * dash::max_element (first, last);
+  ElementType diff = (abs(last_possible_key - first_possible_key) / p) * 0.1;
+
+  dash::barrier();
+
+  if (p == 1) {
+    return std::sort(first, last);
```

Comment:
Do this before the setup above.

Reply:
And also check if the whole range resides on one unit:

```cpp
if(first.is_local() && (last-1).is_local()){
  return std::sort(first.local(), (last-1).local()+1);
}
// + tell or check on the other units that they can return without sorting
```
the -1 trick is necessary as the global last iterator might point to the next unit, but `sort()` takes a past-end iterator, hence this iterator does not have to be dereferenced.